test/evergreen/find_gperftools.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+# Currently evergreen machine doesn't contain gperftools binaries. The aim of this script is to
+# create a caching mechanism for fetching gperftools version 2.9.1. First we check if the binaries
+# for the machine are present in 10gen S3 bucket. If not, we download the sources, build the
+# binaries and upload the binaries to 10gen S3 Bucket.
```

Comment:
I'd expect we have a finite number of machine types to support. Why not pre-build those binaries and push them to s3 bucket beforehand, i.e. decoupling binary building and downloading, same as [what clang-format does right now](https://github.com/wiredtiger/wiredtiger/blob/40af161e2601336298f57ce1ddb9b0edd0bc1332/dist/s_clang-format#L10).

Also, I'd suggest we avoid calling aws s3 command directly and passing aws credentials around in the script if possible for security consideration.

Reply:
The clang format binaries are dependent per OS. Building tcmalloc binaries is dependent per build variant, e.g. ubuntu-1804 needs to be built separately from ubuntu-2004. This means every time a new variant is introduced into our evergreen testing, we will need to manually pre-build the binaries for the new variant. The advantage of having building and downloading in the script, is that the process is automated. 

An additional benefit to this design, is to also easily follow mongodb's gperftools version. Since mongodb regularly updates their gperftools, an easy modification of the script allows to follow mongodb's gperftools version.

Reply:
> Also, I'd suggest we avoid calling aws s3 command directly and passing aws credentials around in the script if possible for security consideration.

I agree, when designing the script, I was skeptical of passing around `aws_keys` and `aws_secrets` into the script. Is there a better we can handle aws commands from a script file? Currently I am using `set +o verbose` to hide any leakage of those variables. The benefit of using aws commands within this script allows us to check if a binary exists in the S3 Bucket. Inside `evergreen.yml`, calling s3.get on an non-existent file, fails instead.

Reply:
Thanks for the explanations. 
> every time a new variant is introduced into our evergreen testing, we will need to manually pre-build the binaries for the new variant.

I don't expect we frequently change build variants (maybe once every a few years). 

> An additional benefit to this design, is to also easily follow mongodb's gperftools version. 

I'm not fully convinced mongo server changes update their gperftools regularly. I feel rebuilding the set of binaries every half a year (or every year) is enough for us. Some building scripts will be helpful to achieve a certain level of automation considering it would be a recurring need.

Reply:
Had a talk with @lukech , we will continue with using the same functionality inside gperftools script under automations-script repo. This allows us to be more confident and secure using aws-cli under a private repo.